eng/xplat-job.yml

```diff
@@ -148,6 +148,38 @@ jobs:
       - name: crossPackagesArg
         value: ''
 
+    - name: testArtifactRootName
```

Comment:
Would it be better to have two .yml step templates for Archive-And-Upload-Artifacts and Download-And-Unpack-Artifacts and use them in build-test-job.yml and run-test-job.yml? 

They can hide `binTestsPath`, `testRootFolderPath` and the remaining variables that, in my opinion, don't belong to xplat-job.yml. xplat-job.yml was designed to be a *"base"* class for other jobs and now it has a knowledge about the test directories layout and artifact names which should be known only to test related jobs.

Reply:
I have refactored the new jobs as you suggested. I have added a pair of new step templates "upload-artifact-step.yml" and "download-artifact-step.yml". I tried to make them sufficiently general so that they should be theoretically usable to zip / unzip the entire GitHub enlistment within a particular build pipeline which is one of the other potential tasks I recently discussed with Sergiy and Jarret. I have also added a new job template "xplat-test-job.yml" that specializes the job "xplat-job.yml" by adding the variables used specifically by the test jobs.